### PR TITLE
Libresprite and googletest

### PR DESCRIPTION
**Pull request by Muvkina**
Please fill out the template.

Read: https://k1ss.org/wiki/kiss/style-guide

---

## Description of package
Pixel art program forked from aseprite before aseprite went non oss. Googletest is needed by libresprite

## Installed manifest of package

```
/var/db/kiss/installed/libresprite/version
/var/db/kiss/installed/libresprite/sources
/var/db/kiss/installed/libresprite/manifest
/var/db/kiss/installed/libresprite/depends
/var/db/kiss/installed/libresprite/checksums
/var/db/kiss/installed/libresprite/build
/var/db/kiss/installed/libresprite/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/local/share/libresprite/data/www/index.html
/usr/local/share/libresprite/data/www/aseprite.png
/usr/local/share/libresprite/data/www/aseprite.js
/usr/local/share/libresprite/data/www/aseprite.css
/usr/local/share/libresprite/data/www/
/usr/local/share/libresprite/data/widgets/webp_options.xml
/usr/local/share/libresprite/data/widgets/user_data.xml
/usr/local/share/libresprite/data/widgets/undo_history.xml
/usr/local/share/libresprite/data/widgets/timeline_conf.xml
/usr/local/share/libresprite/data/widgets/sprite_size.xml
/usr/local/share/libresprite/data/widgets/sprite_properties.xml
/usr/local/share/libresprite/data/widgets/send_crash.xml
/usr/local/share/libresprite/data/widgets/select_accelerator.xml
/usr/local/share/libresprite/data/widgets/replace_color.xml
/usr/local/share/libresprite/data/widgets/paste_text.xml
/usr/local/share/libresprite/data/widgets/palette_size.xml
/usr/local/share/libresprite/data/widgets/palette_popup.xml
/usr/local/share/libresprite/data/widgets/palette_from_sprite.xml
/usr/local/share/libresprite/data/widgets/options.xml
/usr/local/share/libresprite/data/widgets/new_sprite.xml
/usr/local/share/libresprite/data/widgets/new_layer.xml
/usr/local/share/libresprite/data/widgets/new_folder_window.xml
/usr/local/share/libresprite/data/widgets/modify_selection.xml
/usr/local/share/libresprite/data/widgets/main_window.xml
/usr/local/share/libresprite/data/widgets/layer_properties.xml
/usr/local/share/libresprite/data/widgets/keyboard_shortcuts.xml
/usr/local/share/libresprite/data/widgets/jpeg_options.xml
/usr/local/share/libresprite/data/widgets/import_sprite_sheet.xml
/usr/local/share/libresprite/data/widgets/home_view.xml
/usr/local/share/libresprite/data/widgets/grid_settings.xml
/usr/local/share/libresprite/data/widgets/goto_frame.xml
/usr/local/share/libresprite/data/widgets/gif_options.xml
/usr/local/share/libresprite/data/widgets/frame_tag_properties.xml
/usr/local/share/libresprite/data/widgets/frame_properties.xml
/usr/local/share/libresprite/data/widgets/font_popup.xml
/usr/local/share/libresprite/data/widgets/file_selector.xml
/usr/local/share/libresprite/data/widgets/export_sprite_sheet.xml
/usr/local/share/libresprite/data/widgets/duplicate_sprite.xml
/usr/local/share/libresprite/data/widgets/despeckle.xml
/usr/local/share/libresprite/data/widgets/convolution_matrix.xml
/usr/local/share/libresprite/data/widgets/color_curve_point.xml
/usr/local/share/libresprite/data/widgets/cel_properties.xml
/usr/local/share/libresprite/data/widgets/canvas_size.xml
/usr/local/share/libresprite/data/widgets/brush_slot_params.xml
/usr/local/share/libresprite/data/widgets/advanced_mode.xml
/usr/local/share/libresprite/data/widgets/
/usr/local/share/libresprite/data/skins/default/skin.xml
/usr/local/share/libresprite/data/skins/default/sheet.png
/usr/local/share/libresprite/data/skins/default/palette.png
/usr/local/share/libresprite/data/skins/default/minifont.png
/usr/local/share/libresprite/data/skins/default/font.png
/usr/local/share/libresprite/data/skins/default/
/usr/local/share/libresprite/data/skins/
/usr/local/share/libresprite/data/scripts/white_to_alpha.js
/usr/local/share/libresprite/data/scripts/
/usr/local/share/libresprite/data/pref.xml
/usr/local/share/libresprite/data/palettes/zx-spectrum.gpl
/usr/local/share/libresprite/data/palettes/x11.gpl
/usr/local/share/libresprite/data/palettes/win16.gpl
/usr/local/share/libresprite/data/palettes/web-safe-colors.gpl
/usr/local/share/libresprite/data/palettes/vga-13h.gpl
/usr/local/share/libresprite/data/palettes/teletext.gpl
/usr/local/share/libresprite/data/palettes/solarized.gpl
/usr/local/share/libresprite/data/palettes/smile-basic.gpl
/usr/local/share/libresprite/data/palettes/psygnork.gpl
/usr/local/share/libresprite/data/palettes/pico-8.gpl
/usr/local/share/libresprite/data/palettes/nes.gpl
/usr/local/share/libresprite/data/palettes/nes-ntsc.gpl
/usr/local/share/libresprite/data/palettes/monokai.gpl
/usr/local/share/libresprite/data/palettes/master-system.gpl
/usr/local/share/libresprite/data/palettes/jmp.gpl
/usr/local/share/libresprite/data/palettes/google-ui.gpl
/usr/local/share/libresprite/data/palettes/gameboy.gpl
/usr/local/share/libresprite/data/palettes/gameboy-color-type1.gpl
/usr/local/share/libresprite/data/palettes/eroge-copper.gpl
/usr/local/share/libresprite/data/palettes/db32.gpl
/usr/local/share/libresprite/data/palettes/db16.gpl
/usr/local/share/libresprite/data/palettes/cpc-boy.gpl
/usr/local/share/libresprite/data/palettes/copper-tech.gpl
/usr/local/share/libresprite/data/palettes/commodore64.gpl
/usr/local/share/libresprite/data/palettes/commodore-vic20.gpl
/usr/local/share/libresprite/data/palettes/commodore-plus4.gpl
/usr/local/share/libresprite/data/palettes/cga.gpl
/usr/local/share/libresprite/data/palettes/cg-arne.gpl
/usr/local/share/libresprite/data/palettes/atari2600-pal.gpl
/usr/local/share/libresprite/data/palettes/atari2600-ntsc.gpl
/usr/local/share/libresprite/data/palettes/arne32.gpl
/usr/local/share/libresprite/data/palettes/arne16.gpl
/usr/local/share/libresprite/data/palettes/arne-paldac.gpl
/usr/local/share/libresprite/data/palettes/apple-ii.gpl
/usr/local/share/libresprite/data/palettes/a64.gpl
/usr/local/share/libresprite/data/palettes/
/usr/local/share/libresprite/data/icons/doc64.png
/usr/local/share/libresprite/data/icons/doc48.png
/usr/local/share/libresprite/data/icons/doc32.png
/usr/local/share/libresprite/data/icons/doc16.png
/usr/local/share/libresprite/data/icons/doc.ico
/usr/local/share/libresprite/data/icons/ase64.png
/usr/local/share/libresprite/data/icons/ase48.png
/usr/local/share/libresprite/data/icons/ase32.png
/usr/local/share/libresprite/data/icons/ase16.png
/usr/local/share/libresprite/data/icons/ase.ico
/usr/local/share/libresprite/data/icons/
/usr/local/share/libresprite/data/gui.xml
/usr/local/share/libresprite/data/fonts/ase2.pcx
/usr/local/share/libresprite/data/fonts/ase1.pcx
/usr/local/share/libresprite/data/fonts/anipro_variable.pcx
/usr/local/share/libresprite/data/fonts/anipro_fixed.pcx
/usr/local/share/libresprite/data/fonts/allegro.pcx
/usr/local/share/libresprite/data/fonts/
/usr/local/share/libresprite/data/convmatr.def
/usr/local/share/libresprite/data/
/usr/local/share/libresprite/
/usr/local/share/
/usr/local/bin/libresprite
/usr/local/bin/
/usr/local/
/usr/
```

## New package

- [ x] Latest upstream version.
- [ x] I agree to maintain this package.
- [ x] I agree to notify @eudaldgr if I can no longer maintain this package.

### CODE REVIEW

**Comment by Muvkina** [comment]
I removed the lcms changes from this pr as they shouldn't have been in here.

File: gfx/libresprite/depends [hunk 1]
```diff
@@ -0,0 +1,8 @@
+googletest
+curl
+openjpeg2
+libpng
+libwebp
+pixman
+giflib
+
```
Comment (eudaldgr) [inline gfx/libresprite/depends:7]:
```suggestion

```

File: gfx/libresprite/version [hunk 2]
```diff
@@ -0,0 +1 @@
+1.1.7 1
```
Comment (eudaldgr) [inline gfx/libresprite/version]: It's git or it's 1.1.7 ??

File: devel/googletest/build [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh -e
+
+export DESTDIR="$1"
+
+cmake -B build \
+  -DCMAKE_INSTALL_PREFIX=/usr \
```
Comment (eudaldgr) [inline devel/googletest/build]:
```suggestion
  -DCMAKE_INSTALL_PREFIX=/usr
```

File: devel/googletest/build [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  -DFLAG=1
```
Comment (eudaldgr) [inline devel/googletest/build]:
```suggestion

```
That's the example on the wiki `-DFLAG=1` means any example flag.

File: devel/googletest/build [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+cmake --build build
```
Comment (eudaldgr) [inline devel/googletest/build]: https://k1ss.org/wiki/kiss/style-guide#0210

File: devel/googletest/build [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh -e
```
Comment (eudaldgr) [inline devel/googletest/build]: [This](https://k1ss.org/wiki/kiss/style-guide#0202) in all the build script.

File: gfx/libresprite/build [hunk 7]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh -e
```
Comment (eudaldgr) [inline gfx/libresprite/build:1]: [This](https://k1ss.org/wiki/kiss/style-guide#0202) in all the build script.

File: gfx/libresprite/build [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+sed -i '/pthread_detach/d' src/base/thread.cpp
+
+git submodule update --init --recursive
+
+export DESTDIR="$1"
+
+mkdir build && cd build
```
Comment (eudaldgr) [inline gfx/libresprite/build:9]:
```suggestion

```

File: gfx/libresprite/build [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+cmake \
```
Comment (eudaldgr) [inline gfx/libresprite/build:10]:
```suggestion
cmake -B build \
```

File: gfx/libresprite/build [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  -DUSE_SHARED_CURL=ON \
+  -DUSE_SHARED_ZLIB=ON \
+  -DUSE_SHARED_JPEGLIB=ON \
+  -DUSE_SHARED_LIBPNG=ON \
+  -DUSE_SHARED_LIBWEBP=ON \
+  -DUSE_SHARED_PIXMAN=ON \
+  -DUSE_SHARED_FREETYPE=ON \
+  -DUSE_SHARED_GIFLIB=ON \
+  -DUSE_SHARED_GTEST=ON \
+  -G Ninja ..
```
Comment (eudaldgr) [inline gfx/libresprite/build:19]:
```suggestion

```

File: gfx/libresprite/build [hunk 11]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh -e
+
+sed -i '/pthread_detach/d' src/base/thread.cpp
+
+git submodule update --init --recursive
+
+export DESTDIR="$1"
```
Comment (eudaldgr) [inline gfx/libresprite/build]: I prefer `export ...` before other things.

File: gfx/libresprite/build [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+cmake \
+  -DUSE_SHARED_CURL=ON \
+  -DUSE_SHARED_ZLIB=ON \
+  -DUSE_SHARED_JPEGLIB=ON \
+  -DUSE_SHARED_LIBPNG=ON \
+  -DUSE_SHARED_LIBWEBP=ON \
+  -DUSE_SHARED_PIXMAN=ON \
+  -DUSE_SHARED_FREETYPE=ON \
+  -DUSE_SHARED_GIFLIB=ON \
+  -DUSE_SHARED_GTEST=ON \
```
Comment (eudaldgr) [inline gfx/libresprite/build]:
```suggestion
  -DUSE_SHARED_GTEST=OFF
```
With this we don't need `googletest`, please remove them.

File: gfx/libresprite/build [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  -G Ninja ..
+
+ninja -C ./ libresprite
```
Comment (eudaldgr) [inline gfx/libresprite/build:21]: Please, follow cmake [style-guide](https://k1ss.org/wiki/kiss/style-guide#0800), it works for me.

**Review by eudaldgr — COMMENTED** [review]
I suggest some changes, please consider them.

**Comment by eudaldgr** [comment]
And if you can, clean the commits, it can be done with only one, rebase/fetch or do wathever it needs.

**Comment by Muvkina** [comment]
Few things. I can change most of these. The only thing is doesn't kiss prefer shared libraries? So googletest should be packaged seperately? If you dont wan't that that is ok I was just wondering. When it comes to the version not being git idk why i didnt write git... I remember I had some problems with the release tar I will mess around with that as well in a few.

**Comment by eudaldgr** [comment]
`googletest` is not necessary. And you can't have a version with '1.1.7 1' and after a source with 'git+https://.....'

**Pull request closed without merging**